tests/liquidityProvider.ts

```diff
@@ -0,0 +1,240 @@
+	eventSubscriber.subscribe();
+
+	let userAccountPublicKey: PublicKey;
+
+	let usdcMint;
+	let userUSDCAccount;
+
+	// ammInvariant == k == x * y
+	const mantissaSqrtScale = new BN(Math.sqrt(MARK_PRICE_PRECISION.toNumber()));
+	const ammInitialQuoteAssetReserve = new anchor.BN(5 * 10 ** 13).mul(
+		mantissaSqrtScale
+	);
+	const ammInitialBaseAssetReserve = new anchor.BN(5 * 10 ** 13).mul(
+		mantissaSqrtScale
+	);
+
+	const usdcAmount = new BN(30 * 10 ** 8);
+
+    let clearingHouseUser: ClearingHouseUser;
+	before(async () => {
+		usdcMint = await mockUSDCMint(provider);
+		userUSDCAccount = await mockUserUSDCAccount(usdcMint, usdcAmount, provider);
+
+		clearingHouse = Admin.from(
+			connection,
+			provider.wallet,
+			chProgram.programId,
+			{
+				commitment: 'confirmed',
+			},
+			0,
+			[new BN(0), new BN(1), new BN(2), new BN(3), new BN(4)],
+			[new BN(0)]
+		);
+		await clearingHouse.initialize(usdcMint.publicKey, true);
+		await clearingHouse.subscribe();
+
+		await initializeQuoteAssetBank(clearingHouse, usdcMint.publicKey);
+		await clearingHouse.initializeMarket(
+			await mockOracle(1),
+			ammInitialBaseAssetReserve,
+			ammInitialQuoteAssetReserve,
+			new BN(0)
+		);
+		
+		[, userAccountPublicKey] =
+			await clearingHouse.initializeUserAccountAndDepositCollateral(
+				usdcAmount,
+				userUSDCAccount.publicKey
+			);
+
+        clearingHouseUser = ClearingHouseUser.from(clearingHouse, provider.wallet.publicKey)
+        clearingHouseUser.subscribe()
+	});
+
+	after(async () => {
+		await clearingHouse.unsubscribe();
+		await eventSubscriber.unsubscribe();
+        await clearingHouseUser.unsubscribe();
+	});
+
+    // adds lp when 
+    //      fresh-init market [x]
+    //      non-fresh-init market 
+	it('provides and removes liquidity', async () => {
+		var market = clearingHouse.getMarketAccount(0);
+		//var market = await chProgram.account.market.fetch(market.pubkey);
+		const prevSqrtK = market.amm.sqrtK
+        const prevbar = market.amm.baseAssetReserve;
+        const prevqar = market.amm.quoteAssetReserve;
+
+        console.log('adding liquidity...')
+        await chProgram.methods
+            .addLiquidity(
+                new BN(100), 
+                new BN(0), 
+            )
+            .accounts({
+                state: await clearingHouse.getStatePublicKey(),
+                user: await clearingHouse.getUserAccountPublicKey(), 
+                authority: clearingHouse.wallet.publicKey, 
+                oracle: clearingHouse.getMarketAccount(0).amm.oracle
+            })
+            .remainingAccounts([{
+                pubkey: market.pubkey, 
+                isSigner: false, 
+                isWritable: true, 
+            }])
+            .rpc()
+
+		//var market = await chProgram.account.market.fetch(market.pubkey);
+		market = clearingHouse.getMarketAccount(0);
+        assert(prevSqrtK.lt(market.amm.sqrtK)) // k increases = more liquidity 
```

Comment:
would be better to have a more precise check of how much k increases as it will make the test more robust to future changes. You should be able to calculate exactly what sqrt k will be after update